src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
+import static java.lang.Math.abs;
+
+import java.util.HashSet;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private HashObject<K, V>[] hashMap;
+    private HashSet<K> keys;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size = 0;
+
+    public MyHashMap() {
+        hashMap = new HashObject[DEFAULT_INITIAL_CAPACITY];
+        keys = new HashSet<K>();
+    }
+
+    public HashSet<K> keySet() {
+        return keys;
+    }
+
+    public boolean containsKey(K key) {
+        return keys.contains(key);
+    }
+
+    public V[] values() {
```

Comment:
The method `values()` is returning an array of type `V[]`, but it is being cast from `Object[]`. This can lead to a `ClassCastException` at runtime. Consider using a different approach to store and return the values.